src/libraries/System.Private.CoreLib/src/System/Numerics/TotalOrderIeee754Comparer.cs

```diff
@@ -142,12 +142,28 @@ static int CompareSignificant(T x, T y)
                         // To match the integer semantic comparison above, here we compare all the significant bits
                         // Revisit this if decimals are added
 
-                        Span<byte> significantX = stackalloc byte[x!.GetSignificandByteCount()];
-                        Span<byte> significantY = stackalloc byte[y!.GetSignificandByteCount()];
-                        x.WriteSignificandBigEndian(significantX);
-                        y.WriteSignificandBigEndian(significantY);
+                        // Leave the space for custom floating-point type that has variable significand length
 
-                        return significantX.SequenceCompareTo(significantY);
+                        if (x!.GetSignificandBitLength() == y!.GetSignificandBitLength())
```

Comment:
nit: Would be good to cache these to avoid a secondary lookup and/or more JIT work in the common case.